Convert state machine diagram to Mermaid

mortenpi opened the pull request:
Converts the state machine diagram in `docs/` to a Mermaid diagram and moves it into the README. In particular, the current diagram is not very readable in dark mode. But also it's just an image, so it's not very maintainable.

![image](https://github.com/JuliaComputing/PkgAuthentication.jl/assets/147757/e489a58b-5a86-4926-8068-fd966317e35d)

So this takes advantage of GitHub's mermaid.js integration, which has an interactive way to move around on the diagram (granted, the UI is not super-great, but...). You should be [able to try out the interactive UI here](https://github.com/mortenpi/PkgAuthentication.jl/blob/93f6b59d132e2edcfae68542ce736550d358b9ff/README.md#implementation).

![image](https://github.com/JuliaComputing/PkgAuthentication.jl/assets/147757/9fa482b2-d400-4348-a3b3-983300b124d4)

I also added "sticky notes" that also explain each of the states of the machine, which I reverse engineered from reading the code.

![image](https://github.com/JuliaComputing/PkgAuthentication.jl/assets/147757/6183f37e-0465-4bfa-a879-2ff9f197742a)

DilumAluthge commented:
I like the new diagram.

I'm not sure about moving it to the README. The current diagram used to be in the README, but I moved it out because I thought it was overwhelming to show it to a new user. I figured that new users probably just need to see how to use the package, and then they can dive into the internal details only if they're interested.

mortenpi commented:
That's a fair point. I brought the `docs/internals.md` page back: https://github.com/mortenpi/PkgAuthentication.jl/blob/mp/state-diagram/docs/internals.md

mortenpi commented:
Done. Moved the notes into docstrings, updated the script, and the `docs/internals.md` is now generated by `bin/structure.jl`. I also added a Pkg environment there so that it would be easier to install the dependencies (well, the one small dependency the updated script has).
